src/activities/game/Game.cpp

```diff
@@ -1092,6 +1093,143 @@ namespace activities::game
 
 	void Game::DrawSampleInfo() const
 	{
+		if (!drawHUD)
+			return;
+
+		auto s = simulationSample;
+		auto pt = s.particle;
+
+		int wavelength = 0;
+		StringBuilder hudText;
+		String type = simulation->ElementResolve(pt.type, pt.ctype, pt.tmp, pt.tmp4);
+
+		if (false /*Lua is used*/)
+		{
+			// TODO-REDO_UI Custom hud in Lua
+			// One more if below checks if lua is used!!!
+		}
+		else
+		{
+			if (s.WallType > 0 && pt.type <= 0)
+				hudText << simulation->wtypes[s.WallType].name;
+			else
+				hudText << type;
+
+			if (pt.type > 0)
+			{
+				uint16_t hudProp = simulation->elements[pt.type].HudProperties;
+
+				if((hudProp & HUD_CTYPE_BITS) == HUD_CTYPE_WAVELENGTH)
+					wavelength = pt.ctype & 0x3FFFFFFF;
+
+				hudText << ", Temp: " << Format::Precision(pt.temp - 273.15, 2) << "C";
+				if (debugHUD)
+				{
+					hudText << ", Life: " << pt.life;
+
+					if ((hudProp & HUD_TMP_BITS) == HUD_TMP_NUM)
+						hudText << ", Tmp: "  << pt.tmp;
+					else if ((hudProp & HUD_TMP_BITS) == HUD_TMP_TYPEORNUM)
+					{
+						if (simulation->IsElement(TYP(pt.tmp)))
+							hudText << ", Tmp: " << simulation->ElementResolve(TYP(pt.tmp), ID(pt.tmp), 0, 0, false);
+						else
+							hudText << ", Tmp: " << pt.tmp;
+					}
+
+					if ((hudProp & HUD_TMP2_BITS) == HUD_TMP2_NUM)
+						hudText << ", Tmp2: " << pt.tmp2;
+				}
+			}
+
+			if (s.isMouseInSim) // Non particle stuff like pressure gravity etc.
+			{
+				hudText << ", Pressure: " << Format::Precision(s.AirPressure,2) << "\n";
+				if (pt.type > 0)
+					hudText << "#" << s.ParticleID << ", ";
+
+				if(newtonianGravity)
+				{
+					hudText << "GX: " << Format::Precision(s.GravityVelocityX, 2);
+					hudText << ", GY: " << Format::Precision(s.GravityVelocityY, 2) << ", ";
+				}
+
+				if(ambientHeat)
+					hudText << "AHeat: " << Format::Precision(s.AirTemperature - 273.15, 2) << "C, ";
+			}
+			else
+				hudText << "\n";
+			hudText << "X:" << s.PositionX << ", Y:" << s.PositionY;
+
+		}
+
+		// TODO-REDO_UI Make the hud move under zoom window if it overlaps, draw filt spectrum
+		auto lines = hudText.Build().PartitionBy("\n");
+
+		int posY = 10;
+		for (int i = 0; i < lines.size(); ++i)
+		{
+			String line = lines[i];
+
+			int paddingH = 4;
+			int paddingV = 3;
+			auto &g = gui::SDLWindow::Ref();
+
+			gui::Point size = g.TextSize(line);
+			gui::Rect rect;
+			rect.pos = { XRES - 10 - size.x , posY };
+			rect.size = size;
+			gui::Rect inflatedRect = rect.Grow({ paddingH, paddingV }).Offset({ -paddingH / 2, -paddingV / 2 });
+
+			g.DrawRect(inflatedRect, { 0, 0, 0, 128 });
+			g.DrawText(rect.Offset({ 1, 1 }).pos, line, { 255, 255, 255, 191 });
+			posY += size.y + paddingV;
+
+			if (wavelength != 0 && i == 0 && (!false/*not using lua*/) )
+			{
+				DrawWavelengthSpectrum(rect.Offset({ -2, -2 }).pos, wavelength);
+			}
+		}
+	}
+
+	void Game::DrawWavelengthSpectrum(gui::Point pos, int wavelength) const
+	{
+		auto &graphics = gui::SDLWindow::Ref();
```

Comment:
Would be nice if you kept with the `g` convention. I'm also guilty of breaking it >_>